source/language/delays.rst

```diff
@@ -287,47 +287,161 @@ to properly take into account the finite length of each gate.
 Boxed expressions
 -----------------
 
-We introduce a ``box`` expression for scoping a particular part of the circuit.
-A boxed subcircuit can never be inlined (until target code generation
-time), and optimizations across the boundary of a box are forbidden. The
-contents inside the box can be optimized. The contents around the box
-can be optimized too, e.g. it is permissible to commute a gate past a
-box by knowing the unitary implemented by the box. Delays that are
-within a box are implementation details of the box; they are invisible
-to the outside scope and therefore do not prevent commutation.
-
-We introduce a ``boxas`` expression for labeling a box. We primarily use this to
-later refer to the length of this box. Boxed expressions are good for
-this because their contents are isolated and cannot be combined with
-gates outside the box. Therefore, no matter how the contents of the box
-get optimized, the ``lengthof(boxlabel`` has a well-defined meaning.
+We introduce a ``box`` statement for scoping the timing of a particular part of the circuit.
+A boxed subcircuit is different from a ``gate`` or ``def`` subroutine, in that it is merely 
+a pointer to a piece of code within the larger scope which constains it. This can be used to
```

Comment:
"pointer" in this description is a bit off, since I don't have a semantic way to refer to a particular `box`.